config/migrations/20220615120000-public-decrees.sparql

```diff
@@ -0,0 +1,73 @@
+PREFIX besluit: <http://data.vlaanderen.be/ns/besluit#>
+PREFIX dossier: <https://data.vlaanderen.be/ns/dossier#>
+PREFIX besluitvorming: <http://data.vlaanderen.be/ns/besluitvorming#>
+PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX dct: <http://purl.org/dc/terms/>
+PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+PREFIX prov: <http://www.w3.org/ns/prov#>
+
+DELETE {
+  GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+    ?document ext:toegangsniveauVoorDocumentVersie ?toegangsniveau .
+  }
+}
+INSERT {
+  GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+    ?document ext:toegangsniveauVoorDocumentVersie <http://themis.vlaanderen.be/id/concept/toegangsniveau/c3de9c70-391e-4031-a85e-4b03433d6266> .
+  }
+}
+WHERE {
+  GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+    ?document a dossier:Stuk .
+    ?document dct:title ?title .
+    ?document ext:toegangsniveauVoorDocumentVersie ?toegangsniveau .
+    ?document ^besluitvorming:geagendeerdStuk/^dct:hasPart/besluitvorming:isAgendaVoor ?meeting .
+    ?meeting besluit:geplandeStart ?plannedStart .
+     FILTER ( ?plannedStart > xsd:dateTime("2016-09-08T00:00:00Z"))
+     FILTER ( regex(?title, ".*DEC.") )
+  }
+  GRAPH <http://mu.semte.ch/graphs/public> {
+    ?toegangsniveau skos:prefLabel ?toegangsniveauLabel .
+    FILTER ( ?toegangsniveauLabel != "Publiek"@nl)
+  }
+}
+
+# This query could insert a new publication activity for each meeting.
+# We could also make a construct and separate this to a migration that is not in the repo.
+# ! the "prov:startedAtTime" needs to be changed to a time in the near future to trigger themis/valvas publication
+
+# INSERT {
+#   GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+#     ?newThemisPublicationURI a ext:ThemisPublicationActivity ;
+#       mu:uuid ?newThemisPublicationuuid ;
+#       prov:startedAtTime "2022-07-01T00:00:00+02:00"^^xsd:dateTime ;
+#       ext:scope "newsitems" ;
+#       ext:scope "documents" ;
+#       prov:used ?meeting .
+#   }
+# } WHERE {
+#   GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+#     ?meeting a besluit:Vergaderactiviteit .
+#     {
+#       SELECT DISTINCT ?meeting (STRUUID() AS ?newThemisPublicationuuid) WHERE {
```

Comment:
Did you verify whether this generates a unique uuid for each new publication-activity? There is [a known issue](https://github.com/openlink/virtuoso-opensource/issues/515) in Virtuoso that this is evaluated only once per query, but [a workaround](https://github.com/openlink/virtuoso-opensource/issues/515#issuecomment-456848368) is available.

Reply:
I have tested locally, I tried using the old workaround `BIND(IF(BOUND(?olduuid), STRUUID(), STRUUID()) as ?newuuid)` but that only resulted in 1 ID being reused.
The `STRUUID()` in the select does make unique id's per meeting.
But caution is advised, maybe it works differently in a migration vs sparql interface query

Reply:
this is a small piece of what the construct of the insert query yields
![image](https://user-images.githubusercontent.com/22245223/174766317-682c2d65-c09d-454b-8af9-3d0212c2d0e9.png)